search-relevancy-test/src/search_relevancy_test/reporter.clj

```diff
@@ -39,8 +73,52 @@
           (when (= -1 result-position)
             (println (ansi/style (format "...............Concept %s was not found in the result set"
                                          concept-id)
-                                 :magenta)))))
-      (if (= (sort items-in-positional-order) items-in-positional-order)
-        (println (ansi/style (format "%s OK" description) :green))
-        (println (ansi/style (str description " FAILED - out of order") :red)
-                 colored-items)))))
+                                 :magenta))))
+        (merge anomaly-test
+               {:pass false
+                :positional-order items-in-positional-order
+                :result-description "Missing data"}))
+      (do
+        (let [pass (= (sort items-in-positional-order) items-in-positional-order)]
+          (if pass
+            (println (ansi/style (format "%s OK" description) :green))
+            (println (ansi/style (str description " FAILED - out of order") :red)
+                     colored-items))
+          (merge anomaly-test
+                 {:pass pass
+                  :result-description (when (not pass) "Out of order")
+                  :positional-order items-in-positional-order
+                  :discounted-cumulative-gain (test-discounted-cumulative-gain items-in-positional-order
+                                                                               (count test-concept-ids))
+                  :reciprocal-rank (reciprocal-rank items-in-positional-order)}))))))
+
+(defn- average
+  "Return the average of results, given results is a collection of numbers"
+  [results]
+  (/ (apply + results) (count results)))
+
+(defn generate-result-summary
+  "Calculate fields for the result summary"
+  [results]
+  (let [failed-results (filter #(= false (:pass %)) results)]
+    {:result-description (format "%s tests passed out of %s"
+                                 (count (filter #(= true (:pass %)) results))
+                                 (count results))
+     :pass (empty? failed-results)
+     :average-dcg (average (remove nil? (map :discounted-cumulative-gain results)))
+     :average-failed-dcg (average (remove nil? (map :discounted-cumulative-gain failed-results)))
+     :mean-reciprocal-rank (double (average (remove nil? (map :reciprocal-rank results))))}))
+
+(defn print-result-summary
+  "Create and print the results summary"
+  [results]
+  (let [result-summary (generate-result-summary results)]
+    (println "---------------------------------------")
+    (println "SUMMARY: ")
+    (println (:result-description result-summary))
+    (println (format "Average discounted cumulative gain: %.3f"
```

Comment:
I want the color of this statement to be somewhere on the scale between green and red based on the exact value of the cumulative gain.